lib/components/component.dart

```diff
@@ -15,6 +18,9 @@ import '../extensions/vector2.dart';
 abstract class Component {
   final EffectsHandler _effectsHandler = EffectsHandler();
 
+  final OrderedSet<Component> children =
+      OrderedSet(Comparing.on((c) => c.priority));
+
```

Comment:
In that case we would have to create another layer in between `Component` and `PositionComponent`, because right now a lot of logic that doesn't have to do with it having a position is handled there, other components will need this logic too.
With this solution the implementer of an extension doesn't have to take into consideration that it could have children? They would already have it for free if they would want to use it, they don't have to override anything.

And as a note, children in this PR doesn't refer to how to position things, it is more generic than that since the `rect` argument is removed from the propagation method. The parts that handle positioning will still live in the `PositionComponent`.